### PR TITLE
feat(goreleaser): added the goreleaser pipelines

### PR DESCRIPTION
**Pull request by k4yt3x**
Tested to be working in a GitLab Golang repo.

### CODE REVIEW

File: gitlab/golang/stages/50-deployment/debrelease.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+variables:
+  DEB_BINARY: "${CI_PROJECT_NAME}_${CI_COMMIT_TAG}_linux_amd64.deb"
+  PACKAGE_REGISTRY_URL: "${CI_API_V4_URL}/projects/${CI_PROJECT_ID}/packages/generic/${CI_PROJECT_NAME}/${CI_COMMIT_TAG}"
+
+delivery:debrelease:
```
Comment (rios0rios0) [inline gitlab/golang/stages/50-deployment/debrelease.yaml:5]: This should be in the folder delivery right?

**Comment by rios0rios0** [comment]
Could you please verify your commit signature?

**Review by rios0rios0 — CHANGES_REQUESTED** [review]
Could you please verify your commit signature?

File: gitlab/golang/stages/50-deployment/debrelease.yaml (hunk 1 above)
Comment (k4yt3x) [reply gitlab/golang/stages/50-deployment/debrelease.yaml:5]: Oh it's supposed to be in deployment. I'll change the name

Comment (k4yt3x) [reply gitlab/golang/stages/50-deployment/debrelease.yaml:5]: We do need the build and release to be in two steps.

**Comment by k4yt3x** [comment]
It is signed.

**Review by rios0rios0 — APPROVED** [review]

**Pull request merged**